src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -530,24 +544,36 @@ private void CompleteFromTaskSynchronouslyStub(object task)
 
             public void CompleteFromTaskSynchronously(Task<T> task)
             {
-                if (task.Status == TaskStatus.RanToCompletion)
+                // AsyncTaskMethodBuilder doesn't give us Try* methods, and the Set methods may throw if the task
```

Comment:
Ah, wait, I think I misunderstood how this method is used... it looks like you call it even when the task may not have completed.  Ignore this comment then.